File: src/numeric/max/max.dart

/*
 *  Get the maximum value of a sequence of values
 */

/// Get the maximum value of a sequence of values
/// 
/// The [values] are the values to get the maximum of
/// 
/// Returns the maximum value
T max<T extends num>(List<T> values) {
    T current = values[0];
    for (var value in values) {
        if (value > current) {
            current = value;
        }
    }
    return current;
}

/// The main entry point to the application
/// 
/// The [arguments] are the command line arguments passed to the application
void main(List<String> arguments) {
    List<int> values = (arguments.length > 0)
        ? [for (var arg in arguments) int.parse(arg)]
        : [15, 10];

    int maxValue = max(values);

    print("values: ${values}");
    print("max: ${maxValue}");
}

File: src/numeric/max/max.kt
/**
 * Gets the maximum value from a list of comparable values.
 * @param values The list of values to find the maximum from.
 * @returns The maximum value in the list.
 */
fun <T : Comparable<T>> max(values: List<T>): T {
    var current = values[0]
    for (test in values) {
        if (test > current) {
            current = test
        }
    }
    return current
}

/**
 * The main entry point to the application
 * 
 * @param args The arguments passed into the command line for this instance
 */
fun main(args: Array<String>) {
    var list = listOf(15, 10)
    if (args.size > 0) {
        list = args.map { arg -> arg.toInt() }
    }

    val maxValue = max(list)

    println("values: $list")
    println("max: $maxValue")
}
